### PR TITLE
[Win] Add support for opening files inside corresponding project

### PR DESCRIPTION
**Pull request by King2500**
Check if the specified file is part of a IDEA project and open the file inside the project, if not already opened.

PhpStorm.exe allows opening projects or files inside projects, by providing the project root folder (additionally).

### CODE REVIEW

File: PhpStorm Protocol (Win)/run_editor.js [hunk 1]
```diff
@@ -10,16 +10,41 @@ var settings = {
 };
 
 
-// don't change anything below this line, unless you're know what you're doing
-var url = WScript.Arguments(0),
-	match = /^pstorm:\/\/open\/\?url=file:\/\/(.+)&line=(\d+)$/.exec(url),
-	editor = '"c:\\' + (settings.x64 ? 'Program Files (x86)' : 'Program Files') + '\\JetBrains\\' + settings.folder_name + '\\bin\\PhpStorm.exe" --line %line% "%file%"';
+// don't change anything below this line, unless you know what you're doing
+var url = WScript.Arguments( 0 ),
```
Comment (aik099) [inline PhpStorm Protocol (Win)/run_editor.js]: Please remove space after `(` and before `)`. Repeat in other places in this PR.

File: PhpStorm Protocol (Win)/run_editor.js [hunk 2]
```diff
@@ -10,16 +10,41 @@ var settings = {
...
+		editor += ' "%project%"';
+	}
+
+	editor += ' --line %line% "%file%"';
+
+	var command = editor.replace( /%line%/g, match[2] )
+						.replace( /%file%/g, file )
+						.replace( /%project%/g, project )
+						.replace( /\//g, '\\' )
+						/*.replace(/\\/g, '\\\\')*/;
```
Comment (aik099) [inline PhpStorm Protocol (Win)/run_editor.js]: Debug commented-out code doesn't really do anything. Please remove this line and line below it.

File: PhpStorm Protocol (Win)/run_editor.js [hunk 3]
```diff
@@ -10,16 +10,41 @@ var settings = {
...
 
 if ( match ) {
-	var	shell = new ActiveXObject('WScript.Shell'),
-		file = decodeURIComponent(match[1]).replace(/\+/g, ' '),
-		command = editor.replace(/%line%/g, match[2]).replace(/%file%/g, file).replace(/\//g, '\\')/*.replace(/\\/g, '\\\\')*/;
 
-	shell.Exec(command);
-	shell.AppActivate(settings.window_title);
+	var shell = new ActiveXObject( 'WScript.Shell' ),
+		fso = new ActiveXObject( 'Scripting.FileSystemObject' ),
```
Comment (aik099) [inline PhpStorm Protocol (Win)/run_editor.js]: Please rename `fso` to `file_system`.

File: PhpStorm Protocol (Win)/run_editor.js [hunk 4]
```diff
@@ -10,16 +10,41 @@ var settings = {
...
+		file = decodeURIComponent( match[1] ).replace( /\+/g, ' ' ),
+		searchPath = file.replace( /\//g, '\\' );
```
Comment (aik099) [inline PhpStorm Protocol (Win)/run_editor.js]: Please rename `searchPath` to `search_path`.

**Comment by aik099** [comment]
Today’s review completed.

**Comment by King2500** [comment]
Done.

You want me to squash/combine the two commits into one?

**Comment by aik099** [comment]
Sure.

**Comment by King2500** [comment]
There you go. :)

**Comment by aik099** [comment]
Thanks, merging.

**Pull request merged**